server/controllers/clubAccountControllers.js

```diff
@@ -11,15 +14,53 @@ module.exports = {
         // TODO
     },
     getProfile: function(req, res){
-        // TODO
+        clubProfile.find( {clubAccountId: req.params.id} )
+            .then(clubprofile => res.json(clubprofile))
+            .catch(err => res.status(422).json(err))
     },
     getById: function(req, res){
-        // TODO
+        clubAccount.findById( {_id: req.params.id} )
+            .then(clubaccount => res.json(clubaccount))
+            .catch(err => res.status(422).json(err))
     },
     getByFirebaseId: function(req, res){
-        // TODO
+        clubAccount.find( {firebaseId: req.params.firebaseId} )
+            .then(clubaccount => res.json(clubaccount))
+            .catch(err => res.status(422).json(err))
     },
     getAll: function(req, res){
-        // TODO
+        //query for pending and sort, query for nonpending and sort
+        var rdata = {};
+        clubAccount.find( {verificationStatus: 'pending'} )
+            .sort( {creationDate: -1, lastUpdateDate: -1} ).then(
+                q1 =>{ 
+                    rdata.pending = q1
+                    clubProfile.find({
+                        verificationStatus: 'accepted',
+                        lastUpdateDate:  {
```

Comment:
according to meeting discussion, 
1. use this code block to create a date "14 days from now" 
```
var date = new Date()
        date.setDate(date.getDate() - config.timeFrame)
```
2. for pending accounts, sort by creationDate *ascending*
3. turn each query into json before adding them to rdata.

Reply:
It might be a good idea to use `Promise.all` here rather than chaining promises.